solutions/fully-configurable/main.tf

```diff
@@ -0,0 +1,253 @@
+#######################################################################################################################
+# Resource Group
+#######################################################################################################################
+
+module "resource_group" {
+  source                       = "terraform-ibm-modules/resource-group/ibm"
+  version                      = "1.1.6"
+  existing_resource_group_name = var.existing_resource_group_name
+}
+
+#######################################################################################################################
+# Cloud Logs
+#######################################################################################################################
+
+locals {
+  prefix            = var.prefix != null ? (var.prefix != "" ? var.prefix : null) : null
+  create_cloud_logs = var.existing_cloud_logs_crn == null
+  cloud_logs_crn    = local.create_cloud_logs ? module.cloud_logs[0].crn : var.existing_cloud_logs_crn
+}
+
+module "cloud_logs" {
+  count                                  = local.create_cloud_logs ? 1 : 0
+  source                                 = "../.."
+  resource_group_id                      = module.resource_group.resource_group_id
+  region                                 = var.region
+  instance_name                          = var.cloud_logs_instance_name
+  plan                                   = "standard" # not a variable because there is only one option
+  resource_tags                          = var.cloud_logs_resource_tags
+  access_tags                            = var.cloud_logs_access_tags
+  retention_period                       = var.cloud_logs_retention_period
+  service_endpoints                      = "public-and-private" # not a variable because there is only one option
+  existing_event_notifications_instances = var.existing_event_notifications_instances
+  data_storage = {
+    logs_data = {
+      enabled         = true
+      bucket_crn      = module.buckets.buckets[local.logs_bucket_name].bucket_crn
+      bucket_endpoint = module.buckets.buckets[local.logs_bucket_name].s3_endpoint_direct
+    },
+    metrics_data = {
+      enabled         = true
+      bucket_crn      = module.buckets.buckets[local.metrics_bucket_name].bucket_crn
+      bucket_endpoint = module.buckets.buckets[local.metrics_bucket_name].s3_endpoint_direct
+    }
+  }
+  logs_routing_tenant_regions   = var.logs_routing_tenant_regions
+  skip_logs_routing_auth_policy = var.skip_logs_routing_auth_policy
+  policies                      = var.logs_policies
+}
+
+#######################################################################################################################
+# COS
```

Comment:
The buckets are created before cloud logs - I would put the COS section above Cloud Logs. try to keep them in same order as dependnecy tree so it is easier to follow